api/routes/posts.js

```diff
@@ -1,130 +1,131 @@
-const express = require('express');
-const User = require('../models/User');
-const Post = require('../models/Post');
-const Comment = require('../models/Comment');
-const authenticated = require('../middleware/withAuth');
+const express = require("express");
+const User = require("../models/User");
+const Post = require("../models/Post");
+const Comment = require("../models/Comment");
+const authenticated = require("../middleware/withAuth");
 
 const app = express();
 // to get all the posts
-app.get('/posts', authenticated, (req, res) => {
-	Post.findAll({
-		attributes: ['topic', 'discription', 'city'],
-		include: [
-			{
-				model: User,
-				as: 'user',
-				attributes: ['firstName', 'lastName', 'email'],
-			},
-		],
-	})
-		.then((data) => {
-			console.log(data);
-			res.status(200).send(data);
-		})
-		.catch((err) => {
-			console.log(err);
-			res.sendStatus(400);
-		});
+app.get("/posts", authenticated, (req, res) => {
+  Post.findAll({
+    attributes: ["topic", "description", "post_city"],
+    include: [
+      {
+        model: User,
+        as: "user",
+        attributes: ["firstName", "lastName", "email"],
+      },
+    ],
+  })
+    .then((data) => {
+      console.log(data);
+      res.status(200).send(data);
+    })
+    .catch((err) => {
+      console.log(err);
+      res.sendStatus(400);
+    });
 });
 
 // to get one post
-app.get('/posts/:id', (req, res) => {
-	const post = req.params;
+app.get("/posts/:id", authenticated, (req, res) => {
+  const post = req.params;
 
-	Post.findOne({
-		where: { post_id: post.id },
-		attributes: ['topic', 'discription', 'city'],
-		include: [
-			{
-				model: User,
-				as: 'user',
-				attributes: ['firstName', 'lastName', 'email'],
-			},
-			{
-				model: Comment,
-				attributes: ['comment_content'],
-			},
-		],
-	})
-		.then((data) => {
-			console.log(data);
-			res.status(200).send(data);
-		})
-		.catch((err) => {
-			console.log(err);
-			res.sendStatus(400);
-		});
+  Post.findOne({
+    where: { post_id: post.id },
+    attributes: ["topic", "description", "city"],
```

Comment:
Here in the attributes you forgot to change "city" to "post_city" as in the get all posts route. Otherwise it give an error.

Reply:
done